lib/galaxy/webapps/galaxy/services/history_contents.py

```diff
@@ -1270,10 +1270,6 @@ def __create_dataset_collection(
             dbkey = payload.dbkey
             copy_required = dbkey is not None
             copy_elements = payload.copy_elements or copy_required
-            if copy_required and not copy_elements:
-                raise exceptions.RequestParameterMissingException(
-                    "copy_elements passed as 'false' but it is required to change specified attributes"
-                )
```

Comment:
I think the right fix is to set `copy_elements = payload.copy_elements`, you should not change the dbkey without making a copy.

Reply:
Isn't `copy_elements = payload.copy_elements or copy_required` enforcing the copy and saving from raising an exception?

Reply:
Actually you are right, this would be unexpected behaviour. I've rebased with your suggestion above.